### ISSUE TITLE
Cannot create branch

### ISSUE THREAD

**Issue by Finkman** [sameersbn/docker-gitlab#1045]
When clicking on "create branch" gitlab is running into a timeout and saying "Error 502".
I am running the latest (8.15.2) in a Docker Container on a Synology DS1515+.
On Docker, I can see the processes running on the container and after clicking on "create branch" I see a new process, which seems to stuck:
ruby /home/gitlab/data/repositories/SignalMaster/PrintEnvironmentSimulator.git/hooks/pre-receive
Any idea?

Thank's

**Comment by Finkman** [comment]
Same behavior when deleting a branch on the gitlab frontend.

Branching and deleting a branch on console works.

**Comment by QuickJack** [comment]
After clicking **New Branch** under **Repository>Branches** and entering a new branch name, the new branch is created correctly in 8.15.2. I cannot find a "create branch" button.

**Comment by Finkman** [comment]
Yes, on 8.15.2 it is labled **New Branch**. Nevertheless, creating and deleting branches causing timeouts and 502 errors (_Whoops, GitLab is taking too much time to respond._) on my installation... and I have no idea what to do next.

My setup is:

GitLab 8.15.2

GitLab Shell 4.1.1

GitLab Workhorse

GitLab API v3

Git 2.11.0

Ruby 2.3.3p222

Rails 4.2.7.1

Mysql2 5.5.53-MariaDB

**Comment by Finkman** [comment]
... still discovering this issue on 8.16.5.
May anybody give me a hint what gitlab tries to do on .../hooks/pre-receive when deleting or creating a branch? I would like to figure out what causes the problem, when using the web-ui. 
When i'd retried to delete a branch, with my updated gitlab (now 8.16.5) the web-ui was again not responding for a couple minutes, and returning with the same branches I had before. So deletion failed.
In the protokoll, gitlab tells me: CRIT reaped unknown pid 554)

Please! I really love the create/delete branch feature on the web-ui. Help would be highliy appreciated and I intend to invite you to a beer on next occasion. :-)

**Comment by sxcsfan** [comment]
Same problom here, and I discovered that I'm not able to add README.md from the web ui, anything involves repo commit would turn into a nginx 502 error page.

**Comment by Finkman** [comment]
I am actually on 9.0.0 and tried to add a README.md. Ran into the same error... 502 error page.

**Comment by JonathanBigspin** [comment]
Same problem for me, actually similar to when I ran into this before in issue #1161 but now I can't seem to fix it.

**Comment by JonathanBigspin** [comment]
I fixed it this time by first upgrading to MariaDB 10 and GitLab 9.3.4 and then setting permissions to GitLab folder via File Station to allow read/write by everyone (just like in #1161). Doesn't seem like a good solution though.

**Comment by sxcsfan** [comment]
@JonathanBigspin thanks for the first possible solution, I will give it a try as soon as possible
